src/Tokenizer/Tokens.php

```diff
@@ -858,14 +858,7 @@ public function insertSlices(array $slices): void
         $this->setSize($oldSize + $itemsCount);
 
         krsort($slices);
-
-        $firstIndex = key($slices);
-
-        if (!\is_int($firstIndex) || $firstIndex > $oldSize) {
-            throw new \OutOfBoundsException(sprintf('Invalid index "%s".', $firstIndex));
```

Comment:
this should be reverted, the keys are unique and sorted, so the highest key is the first one,
if the first one is to big we can stop testing the other and throw an exception
if the first is not to big the other cannot be to big as the keys are sorter so always lower

Reply:
fair point. i will do some renaming to make it more obvious